vmm/src/lib.rs

```diff
@@ -253,10 +253,13 @@ impl Vmm {
 
     fn vm_shutdown(&mut self) -> result::Result<(), VmError> {
         if let Some(ref mut vm) = self.vm {
-            vm.shutdown()
+            vm.shutdown()?;
+            self.vm = None;
```

Comment:
What you're looking for is using `take()`, which will leave `None` and retrieve ownership of the object.
```rust
if let Some(mut vm) = self.vm.take() {
        vm.shutdown()?;
}
```